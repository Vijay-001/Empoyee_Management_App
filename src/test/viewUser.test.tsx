import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import ViewEmployeeDetails from "../containers/viewUser/view";
import renderer from "react-test-renderer";

const mockStore = configureMockStore();
const store = mockStore({});

describe("<ViewEmployee/>", () => {
  const Component = () => (
    <Provider store={store}>
      <ViewEmployeeDetails />
    </Provider>
  );

  it("renders correctly", () => {
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders default state values", () => {
    const { getByTestId } = render(<Component />);
    const addUser = getByTestId("addUserDetails");
    fireEvent.click(addUser);
    expect(addUser).not.toHaveClass("Mui-disabled");
  });
});
