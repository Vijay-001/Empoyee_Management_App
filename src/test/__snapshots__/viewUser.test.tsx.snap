// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewEmployee/> renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <p />
  <p />
  <p />
  <p />
  <p />
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained buttonPrimary"
      data-testid="addUserDetails"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Add Employee
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <p />
  <p />
  <p />
  <p />
  <p />
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
             ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
             firstName
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
             LastName
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
             Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
             Action
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            User data not found
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
  />
</div>
`;
